spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/ChangelogIterator.java

```diff
@@ -83,12 +85,16 @@ private ChangelogIterator(
    * @param changeTypeIndex the index of the change type column
    * @param identifierFieldIdx the indices of the identifier columns, which determine if rows are
    *     the same
+   * @param columnSize the number of columns in the table
    * @return a new {@link ChangelogIterator} instance concatenated with the null-removal iterator
    */
-  public static Iterator<Row> iterator(
-      Iterator<Row> rowIterator, int changeTypeIndex, List<Integer> identifierFieldIdx) {
+  public static Iterator<Row> create(
```

Comment:
What about making the signature as follows?

```
public static Iterator<Row> create(
    Iterator<Row> rowIterator,
    StructType rowType,
    List<String> identifierFields) {

  // compute indices for the change and identifier columns here
}
```

I feel like we accept some low-level details in this method. Instead, I'd accept an iterator, schema, and identifier column names to move the complexity from the procedure here.

Reply:
I believe you should already have this code in the procedure. I'd move it here.

Reply:
That's a good idea. Fixed it in the new commit.